homeassistant/components/frontier_silicon/media_player.py

```diff
@@ -52,8 +58,13 @@ async def async_setup_platform(
 ) -> None:
     """Set up the Frontier Silicon platform."""
     if discovery_info is not None:
+        webfsapi_url = await AFSAPI.get_webfsapi_endpoint(
+            discovery_info["ssdp_description"]
+        )
+        afsapi = AFSAPI(webfsapi_url, DEFAULT_PIN)
+
         async_add_entities(
-            [AFSAPIDevice(discovery_info["ssdp_description"], DEFAULT_PASSWORD, None)],
+            [AFSAPIDevice(None, afsapi)],
```

Comment:
Should this not still be using the SSDP description? (as the current entities are not having a unique ID, it will help with backward compatibility in maintaining the same name, and thus entity ID?)

Reply:
The SSDP description is used to retrieve the URL to connect to. This retrieval-step was moved to AFSAPI.get_webfsapi_endpoint() (line 61-62).

The old code also did not use anything else from the SSDP announcement (while it would be possible to use the "SPEAKER-NAME" to set the name). Do I diverge from the old behaviour to already set the `name`, or do I keep that for another PR?

![image](https://user-images.githubusercontent.com/2150060/167561760-3bfe3815-5233-4254-8177-609ad7a9bedc.png)

Reply:
Maybe I'm missing something, but the `None` passed here, now directly becomes the entity name:

![CleanShot 2022-05-11 at 10 42 46](https://user-images.githubusercontent.com/195327/167808200-6229b8c8-6ecc-48fc-ac38-5644aea13070.png)

![CleanShot 2022-05-11 at 10 43 05](https://user-images.githubusercontent.com/195327/167808298-9f3f9b4b-4209-4522-b9b5-0fa44afd28aa.png)

![CleanShot 2022-05-11 at 10 43 29](https://user-images.githubusercontent.com/195327/167808367-33fbdb17-7d3b-4dc0-9159-34531b69e54c.png)

Previously it used the ssdp_description.

Should this become:

```py

        afsapi = AFSAPI(webfsapi_url, DEFAULT_PIN)
        name = await afsapi.get_friendly_name()

        async_add_entities(
            [AFSAPIDevice(name, afsapi)],
```

That way the name is set from the get-go?

Reply:
As I was asked to remove code quality improvements from this PR, I decided to keep it as close to the old behavior as possible, which meant that the name was set upon the first execution of `async_update()`.

Of course your suggestion makes sense, so I'll include it in the PR. 

Note that I've got a lot more code improvements coming up 🙂